ASC-devops-team/ASC-Devops-Server
# Codebase Enhancements with Docker, Yarn, and Prettier

### Overview
This PR encompasses a set of enhancements including the adoption of Yarn as the package manager, updating the deployment script for Docker, and applying code formatting standards through Prettier.

### Changes
- **Prettier Configuration**:
  - Created `.prettierrc.json` with specified coding style guidelines.
- **Docker Deployment**:
  - Updated `deploy.sh` to streamline Docker container deployment and image pruning.
- **Package Manager Transition**:
  - Removed `package-lock.json` in favor of Yarn, to align with the new package manager.
  - Updated `package.json` to reflect the correct author information and added a repository field.
  - Generated a new `yarn.lock` to ensure consistent dependency resolution.

### Testing
- Verify Prettier runs with the new configuration and applies the defined style.
- Execute the `deploy.sh` script to confirm smooth Docker deployment.
- Ensure Yarn correctly manages dependencies without issues.

### Review Checklist
- Review the `.prettierrc.json` for adherence to project formatting standards.
- Confirm `deploy.sh` changes are operational and efficient.
- Validate removal of `package-lock.json` and proper function of Yarn with the `yarn.lock` file.

### Additional
- Developers need to switch to Yarn and reformat the codebase using the new Prettier settings.
- Ensure Docker daemon is running before executing the updated deployment script.
